acq4/pyqtgraph/exporters/HDF5Exporter.py

```diff
@@ -35,14 +35,25 @@ def export(self, fileName=None):
         dsname = self.params['Name']
         fd = h5py.File(fileName, 'a') # forces append to file... 'w' doesn't seem to "delete/overwrite"
         data = []
-        
+
         appendAllX = self.params['columnMode'] == '(x,y) per plot'
-        for i,c in enumerate(self.item.curves):
+        #print dir(self.item.curves[0])
+        tlen = 0
+        for i, c in enumerate(self.item.curves):
             d = c.getData()
+            if i > 0 and len(d[0]) < tlen:
```

Comment:
change to `if i > 0 and len(d[0]) != tlen` and raise an exception rather than `continue`